vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -718,7 +778,16 @@ def invoke_fused_moe_kernel(A: torch.Tensor,
     grid = lambda META: (triton.cdiv(EM, META['BLOCK_SIZE_M']) * triton.cdiv(
         B.shape[1], META['BLOCK_SIZE_N']), )
 
-    if (use_int8_w8a16 or use_int4_w4a16) and \
+    if use_dg:
+        assert use_fp8_w8a8
+        # Note: we never apply the topk_weights here since it requires
+        # unpermuting and resizing the output.  This goes against the
+        # existing interface as the `mul_routed_weight` argument is
+        # ignored.  The weights are applied in _moe_unpermute.
```

Comment:
should this be `The weights are applied in _moe_unpermute_and_reduce`?

Reply:
Yeah, I forgot to update the comment when I renamed the function.